### PR TITLE
[Feature] Support MaxCompute Table for Spark Load

### PR DESCRIPTION
**Pull request by wuxueyang96**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

MaxCompute is a data warehouse solution developed by Alibaba Cloud and has a large number of users.

Currently, users usually use DataX when they want to import data from MaxCompute into StarRocks. If users want to build a global dictionary at the same time, the DataX cannot be satisfied.

While Spark Load of StarRocks can directly help users build a global dictionary, Spark can also access MaxCompute through its own DataSource mechanism. But Spark Load in StarRocks cannot use Spark's Data Source mechanism, it only allows Hive tables to be read using Spark.

This PR extends the capabilities of StarRocks Spark Load by modifying the way Spark Load is written, and allows other Data Sources to access.

### CODE REVIEW

**Comment by caneGuy** [comment]
@stephen-shelby PTAL thanks, actually we need discuss if we should add an OdpsResource or not

**Comment by wuxueyang96** [comment]
run starrocks_fe_unittest

**Comment by caneGuy** [comment]
@wyb @chaoyli PTAL thanks

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveResource.java [hunk 1]
```diff
@@ -28,11 +29,18 @@
...
 
+    private static final String MAX_COMPUTE_ENABLE = "maxcompute.enable";
     // require only one property currently
     private static final String HIVE_METASTORE_URIS = "hive.metastore.uris";
 
+    @SerializedName(value = "isMaxComputeResource")
+    private boolean isMaxComputeResource = false;
+
+    @SerializedName(value = "resourceOptions")
+    private final Map<String, String> resourceOptions = new HashMap<>();
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HiveResource.java:40]: Do the parameters in resourceOptions needn't be verified?

Comment (wuxueyang96) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveResource.java:40]: I'm not sure whether the parameters in resourceOptions need be verified or not. We should verified some definite parameter like `hive.metastore.uris`, but the parameters for spark data source is indefinite.

For example:

If we want to connect to MaxCompute throught Spark, we could use the data source provided by [aliyun-emapreduce-datasources](https://github.com/aliyun/aliyun-emapreduce-datasources) which we must judge `accessKeyId `, we could also use the data source provided by [aliyun-maxcompute-data-collectors](https://github.com/aliyun/aliyun-maxcompute-data-collectors) which we must judge `spark.hadoop.odps.access.id`.

So, for me, I think it's better to judge the resourceOptions by spark data source.

**Comment by wuxueyang96** [comment]
run starrocks_fe_unittest

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/GlobalDictBuilder.java [hunk 2]
```diff
@@ -141,71 +154,84 @@ public GlobalDictBuilder(MultiValueMap dictColumn,
...
     public void createHiveIntermediateTable() throws AnalysisException {
-        Map<String, String> sourceHiveTableColumn = spark.catalog()
-                .listColumns(sourceHiveDBTableName)
-                .collectAsList()
-                .stream().collect(Collectors.toMap(Column::name, Column::dataType));
-
-        Map<String, String> sourceHiveTableColumnInLowercase = new HashMap<>();
-        for (Map.Entry<String, String> entry : sourceHiveTableColumn.entrySet()) {
-            sourceHiveTableColumnInLowercase.put(entry.getKey().toLowerCase(), entry.getValue().toLowerCase());
-        }
```
[the remark this answers is not in the document]
Comment (wuxueyang96) [reply fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/GlobalDictBuilder.java:160]: It was not removed. I think this code only want to get `sourceHiveTableColumn`'s lower case name and value. I did at below:

`String columnType = sourceHiveTableColumn.get(columnName).typeName().toLowerCase();`

but yes, there has a bug. I will fix.

**Comment by wuxueyang96** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 24 / 42 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|5|11|45.45%|[178, 179, 180, 182, 371, 451]|
|:large_blue_circle:|com/starrocks/catalog/HiveResource.java|19|31|61.29%|[57, 61, 62, 72, 73, 74, 75, 88, 103, 104, 107, 114]|

**Comment by wuxueyang96** [comment]
@wyb @stephen-shelby @caneGuy PTAL

**Review by wyb — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Pull request closed without merging**